docs/unified-test-documentation/dasharo-security/200-tpm-support.md

```diff
@@ -404,3 +404,374 @@ Example output:
+    ```
+
+## TPM005.001 ENCRYPT And DECRYPT Functions (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that ENCRYPT and DECRYPT functions are working
+properly.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install `tpm2-tools` using:
+
+    ```bash
+    sudo apt-get install tpm2-tools
+    ```
+
+    [Building from source](https://tpm2-tools.readthedocs.io/en/latest/INSTALL/)
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into the system by using the proper login and password.
+1. Execute commands in the terminal:
+
+    ```bash
+    echo "my secret" > secret.dat
+    sudo tpm2_encryptdecrypt -c key.ctx -o secret.enc secret.dat
+    sudo tpm2_encryptdecrypt -d -c key.ctx -o secret.dec secret.enc
+    ```
+
+1. Read `secret.dec` contents:
+
+    ```bash
+    cat secret.dec
+    ```
+
+1. Note the result
+
+**Expected result**
+
+1. Command should output:
+
+    ```bash
+    my secret
+    ```
+
+## TPM006.001 PCREVENT Function (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that PCREVENT function works properly.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install `tpm2-tools` using:
+
+    ```bash
+    sudo apt-get install tpm2-tools
+    ```
+
+    [Building from source](https://tpm2-tools.readthedocs.io/en/latest/INSTALL/)
+
+**Test steps**
+
+1. Power on the DUT.
+1. Boot into the system.
+1. Log into the system by using the proper login and password.
+1. Execute command in the terminal:
+
+    ```bash
+    echo "foo" > data
+    ```
+
+1. Then hash the `data` using:
+
+    ```bash
+    sudo tpm2_pcrevent data
+    ```
+
+1. Note the result
+
+**Expected result**
+
+1. If process suceeds, last command should give similar output:
+
+    ```bash
+    sha1: f1d2d2f924e986ac86fdf7b36c94bcdf32beec15
+    sha256: b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c
+    sha384: 8effdabfe14416214a250f935505250bd991f106065d899db6e19bdc8bf648f3ac0f1935c4f65fe8f798289b1a0d1e06
+    sm3_256: 2b14a1fc49869413b0beb707069cffc0c6b0a51f3fedb9ce072c80709652b3ae
+    ```
+
+## TPM007.001 PCREXTEND And PCRRESET Functions (Ubuntu 22.04)
+
+**Test description**
+
+This test aims to verify that PCREXTEND and PCRRESET functions are working
+properly.
+
+**Test configuration data**
+
+1. `FIRMWARE` = Dasharo
+1. `OPERATING_SYSTEM` = Ubuntu 22.04
+
+**Test setup**
+
+1. Proceed with the
+    [Test cases common documentation](#test-cases-common-documentation) section.
+1. Install `tpm2-tools` using:
+
+    ```bash
+    sudo apt-get install tpm2-tools
+    ```
+
+    [Building from source](https://tpm2-tools.readthedocs.io/en/latest/INSTALL/)
```

Comment:
What exactly is this link responsible for?

Reply:
Updated